### ISSUE TITLE
Allow usage of external fypp program

### ISSUE THREAD

**Issue by mbanck** [ghb24/NECI_STABLE#5]
It seems NECI insists on the git submodule being present, even if `fypp` is installed in `/usr/local/bin` e.g.:

```
CMake Error at cmake/neci_add_library.cmake:121 (message):
  fypp preprocessor missing.  Please do `git submodule update --init` in the
  git repository.
```
It'd be nice if NECI could use a system-installed version.

Also, the fypp commit id is pretty old, there has been a 3.0 release of fypp this year, is that one working as well?

**Comment by mcocdawc** [comment]
> It'd be nice if NECI could use a system-installed version.

I absolutely agree.

> Also, the fypp commit id is pretty old, there has been a 3.0 release of fypp this year, is that one working as well?

Thank you for the information. There was no specific reason that we did not update it yet, so we should 


PS: Since we mirror our closed developer repository only, if the algorithms to new implementations are published, it will take a while until it shows up here.

**Comment by mcocdawc** [comment]
> It'd be nice if NECI could use a system-installed version.

Implemented now. (It will unfortunately take some time, before we move code to the public repo.)

> Also, the fypp commit id is pretty old, there has been a 3.0 release of fypp this year, is that one working as well?

I tested both 3.0 and `master` and both work. Note that fypp 3.0 still uses `/usr/bin/env python` in its shebang line, which does not work on newer systems where python2 is deprecated. Only the current `master` uses explicitly `/usr/bin/env python3`, so we actually had to update. ;-)
